wallet-new/src/Cardano/Wallet/Kernel/Submission/Worker.hs

```diff
@@ -0,0 +1,41 @@
+module Cardano.Wallet.Kernel.Submission.Worker (
+    tickSubmissionLayer
+    ) where
+
+import           Universum
+
+import           Control.Concurrent (threadDelay)
+import qualified Data.Set as Set
+import           Formatting (sformat, (%))
+import qualified Formatting as F
+
+import           Cardano.Wallet.Kernel.Submission (Evicted, WalletSubmission)
+
+import           System.Wlog (Severity (..))
+
+
+tickSubmissionLayer :: forall m. (MonadCatch m, MonadIO m)
+                    => (Severity -> Text -> m ())
+                    -- ^ A logging function
+                    -> IO (Evicted, WalletSubmission IO)
```

Comment:
Let's just have a single callback here of type `IO ()`. That callback then is responsible for calling `tick`, calling `cancel` on any evicted transactions, and update the state. Note that currently there is a bug in this implementation where we store the new state only if there are evicted transactions.